File: example/src/middleware/response.ts
/**
 * Response Middleware
 *
 * Utilities for formatting Lambda responses with proper headers and CORS
 */

import type { APIGatewayProxyStructuredResultV2 } from 'aws-lambda';

/**
 * Standard JSON response with CORS headers
 */
export const json = (code: number, data: unknown): APIGatewayProxyStructuredResultV2 => {
	return {
		statusCode: code,
		headers: {
			'Content-Type': 'application/json',
			'Access-Control-Allow-Origin': '*', // CORS - adjust for production
			'Access-Control-Allow-Headers': 'Content-Type,Authorization',
			'Access-Control-Allow-Methods': 'GET,POST,PUT,OPTIONS',
		},
		body: JSON.stringify(data),
	};
};

/**
 * CORS preflight response
 */
export const corsPreflightResponse = (): APIGatewayProxyStructuredResultV2 => {
	return {
		statusCode: 204,
		headers: {
			'Access-Control-Allow-Origin': '*',
			'Access-Control-Allow-Headers': 'Content-Type,Authorization',
			'Access-Control-Allow-Methods': 'GET,POST,PUT,OPTIONS',
		},
		body: '',
	};
};

File: src/middleware/response.py
"""Response helper utilities."""

import json
from typing import Any


def json_response(
    status_code: int, body: Any, headers: dict[str, str] | None = None
) -> dict[str, Any]:
    """
    Create a JSON response for AWS Lambda.

    Args:
        status_code: HTTP status code
        body: Response body (will be JSON serialized)
        headers: Optional additional headers

    Returns:
        Lambda proxy response dict
    """
    response_headers = {
        "Content-Type": "application/json",
        "Access-Control-Allow-Origin": "*",
        "Access-Control-Allow-Methods": "GET, POST, OPTIONS",
        "Access-Control-Allow-Headers": "Content-Type, Authorization",
    }

    if headers:
        response_headers.update(headers)

    return {
        "statusCode": status_code,
        "headers": response_headers,
        "body": json.dumps(body, ensure_ascii=False),
    }


def cors_preflight_response() -> dict[str, Any]:
    """Return CORS preflight response for OPTIONS requests."""
    return {
        "statusCode": 200,
        "headers": {
            "Access-Control-Allow-Origin": "*",
            "Access-Control-Allow-Methods": "GET, POST, OPTIONS",
            "Access-Control-Allow-Headers": "Content-Type, Authorization",
        },
        "body": "",
    }
